File: 982-minimum-increment-to-make-array-unique/minimum-increment-to-make-array-unique.c
int comp(const void *a, const void *b){
    return (*(int *)a - *(int *)b);
}
int minIncrementForUnique(int* nums, int numsSize) {
    int result = 0;
    qsort(nums, numsSize, sizeof(int), comp);
    int i = 1;
    for(i; i < numsSize; i++){
        if(nums[i] > nums[i - 1]){
            continue;
        }
        else if(nums[i] == nums[i - 1]){
            nums[i]++;
            result++;
        } else {
            int temp = abs(nums[i] - nums[i - 1]);
            result += temp + 1;
            nums[i] += temp + 1;
        }
    }

    return result;
}

File: 982-minimum-increment-to-make-array-unique/minimum-increment-to-make-array-unique.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var minIncrementForUnique = function(nums) {
    nums.sort((a,b) => a - b);
    let result = 0;

    for(let i = 1; i < nums.length; i++){
        if(nums[i] == nums[i - 1]){
            nums[i]++;
            result++;
        } else if(nums[i] < nums[i - 1]){
            let temp = Math.abs(nums[i] - nums[i - 1]);
            nums[i] += temp + 1;
            result += temp + 1;
        }
    }
    return result;
};